aws/resource_aws_cloudwatch_log_stream.go

```diff
@@ -60,7 +62,24 @@ func resourceAwsCloudWatchLogStreamRead(d *schema.ResourceData, meta interface{}
 
 	group := d.Get("log_group_name").(string)
 
-	ls, exists, err := lookupCloudWatchLogStream(conn, d.Id(), group, nil)
+	var ls *cloudwatchlogs.LogStream
+	var exists bool
+
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		var err error
+		ls, exists, err = lookupCloudWatchLogStream(conn, d.Id(), group, nil)
+		if err != nil {
+			return resource.NonRetryableError(err)
```

Comment:
Maybe it would be good to retry for other kinds of error, too - not sure. For the cases we've seen so far that we'd benefit from retrying on, the initial lookup returns a 200 OK so not retrying on failures would be backward compatible.